src/server/pps/server/api_server.go

```diff
@@ -2171,22 +2201,25 @@ func (a *apiServer) CreatePipeline(ctx context.Context, request *pps.CreatePipel
 		Reprocess:                 request.Reprocess,
 		DryRun:                    request.DryRun,
 	}
-	if _, err := a.createPipeline(ctx, v2Req); err != nil {
+	if err := a.txnEnv.WithWriteContext(ctx, func(txnCtx *txncontext.TransactionContext) error {
+		_, err := a.createPipelineInTransaction(ctx, txnCtx, v2Req)
+		return err
+	}); err != nil {
 		return nil, err
 	}
 	return &emptypb.Empty{}, nil
 }
 
-func (a *apiServer) createPipeline(ctx context.Context, req *pps.CreatePipelineV2Request) (wrappedJSON string, err error) {
-	clusterDefaultsResponse, err := a.GetClusterDefaults(ctx, &pps.GetClusterDefaultsRequest{})
-	if err != nil {
-		return "", status.Error(codes.Internal, "could not get cluster defaults")
+func (a *apiServer) createPipelineInTransaction(ctx context.Context, txnCtx *txncontext.TransactionContext, req *pps.CreatePipelineV2Request) (wrappedJSON string, err error) {
+	var clusterDefaults ppsdb.ClusterDefaultsWrapper
```

Comment:
I think we should move all the Txn relevant logic here to the top of `a.CreatePipelineInTransaction()`, so when pipelines are updated/created inside a transaction, this logic applies